### ISSUE TITLE
Update import to not show warning with Lit 2

### ISSUE THREAD

**Issue by web-padawan** [bennypowers/lit-css-loader#3]
Currently there is a warning when using this loader with Lit 2:

```
The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'."
```

We should change the loader to use `'lit-element/lit-element.js'` which does not trigger the warning for Lit 2.
- https://github.com/lit/lit/blob/main/packages/lit-element/src/lit-element.ts
- https://github.com/lit/lit-element/blob/master/src/lit-element.ts

**Comment by Artur-** [comment]
Not sure why, but in my current project I don't get a warning, I get an error because `node_modules/lit-element` does not exist. Instead there is `node_modules/lit/node_modules/lit-element`

I would suggest a separate Lit 2 version that imports from `'lit'` instead, to avoid any issues

**Comment by web-padawan** [comment]
I don't think we need a separate Lit 2 version. The `lit` package has the following [export](https://github.com/lit/lit/blob/8e3d6ea25cfb86ad18b508f3264856546895f255/packages/lit/src/index.ts#L13):

```js
export * from 'lit-element/lit-element.js';
```

This is exactly what my PR suggests. And it is compatible with both versions of `lit-element`.

**Comment by Artur-** [comment]
It does not work to import from `lit-element/anything` in a project that does not have `lit-element` in `node_modules`. You get there by doing e.g.
```
npm init && npm i --save lit-element@2 && npm i --save lit@2.0.0-rc.1 && npm uninstall lit-element
```

**Comment by web-padawan** [comment]
Confirmed the above issue. Would it help to add `peerDependencies` to the package.json of `lit-css-loader`?
I wonder if npm 7 which [handles peerDependencies automatically](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies) would help in this case.

**Comment by bennypowers** [comment]
We can try the `peerDependencies` approach

**Comment by Haprog** [comment]
I just noticed this discussion here after [commenting](https://github.com/bennypowers/lit-css-loader/pull/4) on the PR.

I'd suggest adding a configuration option to `lit-css-loader` for being able to override the import path of `css`. Then the user could set the import path to `lit` if they're already depending on `lit` (Lit 2), without the need to add also `lit-element` as a direct dependency to the user's project. This would also make the loader compatible with _any_ package that exports a compatible `css` function without requiring any specific dependency.

This could still default to `import { css } from 'lit-element/lit-element.js';` in case no options are given in webpack config (should be backwards compatible?).

**Comment by bennypowers** [comment]
@Haprog [such a thing already exists](https://github.com/bennypowers/rollup-plugin-lit-css#usage-with-fast), with the small caveat that maybe this package needs to be updated to forward the config

Edit: exists on rollup-plugin-lit-css

So, three approaches:

1. reimplement that logic here
2. use some kind of rollup-plugin-to-loader converter (if one exists)
3. export core logic from `rollup-plugin-lit-css` and use it here

**Comment by Haprog** [comment]
@bennypowers Any of those 3 options sounds good to me. They all would have the same user facing API and UX right? So the choice would mainly be about implementation details (what's good for maintenance etc), right?

**Comment by bennypowers** [comment]
yeah seems to me

I'm partial to 3.

I also think it's worth considering setting the default to `lit-element/lit-element.js` and adding the `peerDependency`
**pro**: most users won't need any config
**con**: will install a useless package for non-lit-element users.

PR would ideally close #5 as well 🙏

**Comment by web-padawan** [comment]
Let's discuss how we want to proceed. Here are few PRs that address this issue:

1. Add `import` config option as requested above to allow using `lit` (#7)
2. Use `lit-element/lit-element.js` and add `lit-element` to peerDependencies (#4)
3. Upgrade to use `lit` and make a major release with breaking change (#6)

Which is the preferred solution? I guess the first one, based on the comments.

If you think this is the way to go, I can think about adding some tests for it.

**Comment by bennypowers** [comment]
https://github.com/bennypowers/lit-css-loader/pull/7#issuecomment-846544761

This is published under [v0.1.0](https://www.npmjs.com/package/lit-css-loader/v/0.1.0)

Thank you for contributing 🏆
